lib/commons/dom/get-element-stack.js

```diff
@@ -0,0 +1,488 @@
+/* global dom */
+
+// split the page cells to group elements by the position
+const gridSize = 200; // arbitrary size, increase to reduce memory (less cells) use but increase time (more nodes per grid to check collision)
+
+// https://developer.mozilla.org/en-US/docs/Web/CSS/Replaced_element
+const replacedElements = ['iframe', 'video', 'embed', 'img'];
+
+/**
+ * Determine if node produces a stacking context.
+ * References:
+ * https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context
+ * https://github.com/gwwar/z-context/blob/master/devtools/index.js
+ * @param {VirtualNode} vNode
+ * @return {Boolean}
+ */
+function isStackingContext(vNode) {
+	const node = vNode.actualNode;
+
+	//the root element (HTML)
+	if (
+		!node ||
+		node.nodeName === 'HTML' ||
+		node.nodeName === '#document-fragment'
+	) {
+		return true;
+	}
+
+	const computedStyle = vNode.computedStyle;
+
+	// position: fixed or sticky
+	if (
+		computedStyle.getPropertyValue('position') === 'fixed' ||
```

Comment:
Lookups for the same property value should probably be cached. See https://jsperf.com/get-property-value